widgets/refinement-list.js

```diff
@@ -51,17 +57,30 @@ function refinementList({
     throw new Error(usage);
   }
 
+  if (templates !== defaultTemplates) {
+    templates = Object.assign({}, defaultTemplates, templates);
```

Comment:
What is the `if` for? Can't we always `Object.assign({}, defaultTemplates, templates);`?